### PR TITLE
Fix Annotations and LegoAnimActor class structure.

### PR DESCRIPTION
**Pull request by tahg**
Fix all SIZE annotations for the legoomni module, except for LegoPathBoundary whose constructor is unknown.

Fix all class declarations for LegoAnimActor and its subclasses, including all constructors and virtual functions (declared and stubbed or implemented if small). Yet to do is check overrides for LegoPathActor and its ancestors for these classes.

Add placeholder VTable annotations to LegoAnimActor and subclasses, until vtable.py can handle virtual classes/vbtables.

### CODE REVIEW

File: LEGO1/lego/legoomni/include/legoanimactor.h [hunk 1]
```diff
@@ -4,10 +4,39 @@
...
+	virtual MxResult FUN_1001c1f0(float& p_out);
+	virtual MxResult FUN_1001c360(float, undefined4);
+	virtual MxResult FUN_1001c450(undefined4, undefined4, undefined4, undefined4);
+	virtual void FUN_1001c800();
+
+	// SYNTHETIC: LEGO1 0x1000fb50
+	// LegoAnimActor::`scalar deleting destructor'
+
+private:
+	vector<void*> m_vector; // 0x00
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/include/legoanimactor.h:38]: A vector is `0x10` in size. The offsets aren't correct either. `cvdump` shows `0x08` for the vector.

File: LEGO1/lego/legoomni/include/act3shark.h [hunk 2]
```diff
@@ -3,9 +3,15 @@
 
 #include "legoanimactor.h"
 
-// VTABLE: LEGO1 0x100d7920
+/*
+	VTABLE : LEGO1 0x100d7920 LegoPathActor
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/include/act3shark.h:7]: Please name it either `VTABLE : ` or `VTABLE:`

File: LEGO1/lego/legoomni/include/act2actor.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#include "legoanimactor.h"
+
+/*
+	VTABLE: LEGO1 0x100d6078 LegoPathActor
+	VTABLE: LEGO1 0x100d6148 LegoAnimActor
+*/
+// SIZE 0x1a8
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/include/act2actor.h:7]: Please add `DECOMP_SIZE_ASSERT`s for all the classes.

File: LEGO1/lego/legoomni/include/legoracecar.h [hunk 4]
```diff
@@ -31,8 +39,19 @@ class LegoRaceCar : public LegoCarRaceActor {
 	void VTable0x98() override;                        // vtable+0x98
 	void VTable0x9c() override;                        // vtable+0x9c
 
+	virtual void FUN_10012ea0(float p_worldSpeed);
+	virtual void FUN_10012ff0(float);
+	virtual MxBool FUN_10013130(float);
+
 	// SYNTHETIC: LEGO1 0x10014230
 	// LegoRaceCar::`scalar deleting destructor'
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/include/legoracecar.h:47]: Member should be private (or protected)

File: LEGO1/lego/legoomni/include/legoracecar.h [hunk 5]
```diff
@@ -31,8 +39,19 @@ class LegoRaceCar : public LegoCarRaceActor {
...
+	byte m_unk0x54;           // 0x54
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/include/legoracecar.h:48]: Is there a particluar reason to use `byte`? We use `undefined` for an undefined byte.

File: LEGO1/lego/legoomni/include/act2actor.h [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#include "legoanimactor.h"
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/include/act2actor.h:1]: Please add a include guard.

File: LEGO1/lego/legoomni/src/paths/legoextraactor.cpp [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#include "legoextraactor.h"
+
+// STUB : LEGO1 0x1002a500
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/src/paths/legoextraactor.cpp:3]: There shouldn't be a space here after `STUB`

**Comment by tahg** [comment]
All class structures in legoomni except LegoPathBoundary and LegoObjectFactory should now be the correct size, and guarded by assertions.

File: LEGO1/lego/legoomni/include/legoracemap.h [hunk 8]
```diff
@@ -0,0 +1,35 @@
+#ifndef LEGORACEMAP_H
+#define LEGORACEMAP_H
+
+#include "legoraceactor.h"
+
+/*
+	VTABLE: LEGO1 0x100d5510 LegoRaceActor
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/include/legoracemap.h:7]: 3 classes have the same vtable address?

Comment (tahg) [reply LEGO1/lego/legoomni/include/legoracemap.h:7]: Nope, All virtual classes should have completely different tables, even if nothing is overridden. Technically the compiler could coalesce them, but it doesn't. I'll doublecheck them all.

File: LEGO1/lego/legoomni/include/act3shark.h [hunk 9]
```diff
@@ -3,9 +3,15 @@
 
 #include "legoanimactor.h"
 
-// VTABLE: LEGO1 0x100d7920
+/*
+	VTABLE: LEGO1 0x100d7920 LegoPathActor
+	VTABLE: LEGO1 0x100d79f0 LegoAnimActor
+*/
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/include/act3shark.h:9]: Act3Shark doesn't have a vtable, even though it adds a new virtual function?

Comment (tahg) [reply LEGO1/lego/legoomni/include/act3shark.h:9]: It does, it extends LegoAnimActor non-virtually. I thought it was weird too, but it does in fact use the parent name. (You can verify the list of vtables in cvdump)

**Pull request merged**